### PR TITLE
[#6043] [android] fix "Fetching messages..." shifted down when appears in public chat modal

### LINKED ISSUE #6043: "Fetching messages..." shifted down when appears in public chat modal [Android]
***Type***: Bug

***Summary***: When `Fetching messages...` appears in public chat modal screen it should be placed right below the app top navigation bar. But it's not at the moment. Please see screenshot below.

#### Expected behavior
`Fetching messages...` appears in public chat modal screen right below the app top navigation bar
#### Actual behavior
screen shot 2018-09-27 at 19 03 30

### Reproduction
- Open Status and create new account
- Open any Dapp from dapp list
- Tap modal public chat icon

### Additional Information
* Status version: Develop 0.9.29 (9344)
* Operating System: Android 6, 7, 8

### PR DESCRIPTION
**Pull request by rasom**
fix #6043
status: ready

### CODE REVIEW

File: src/status_im/ui/components/connectivity/view.cljs [hunk 1]
```diff
@@ -21,9 +37,10 @@
...
-                     :accessibility-label :connection-status-text}
-         [react/text {:style    styles/text
-                      :on-press (when mailserver-error?
-                                  #(re-frame/dispatch [:inbox/reconnect]))}
-          (i18n/label label)]]))))
+        [error-label
+         {:view-id           view-id
+          :top               top
+          :window-width      window-width
+          :pending           pending?
```
Comment (dmitryn) [inline src/status_im/ui/components/connectivity/view.cljs]: Seems like missed `?`
Should be ```:pending?```

**Review by dmitryn — APPROVED** [review]

File: src/status_im/ui/components/connectivity/view.cljs (hunk 1 above)
Comment (rasom) [reply src/status_im/ui/components/connectivity/view.cljs]: you are right, will fix tomorrow

Comment (rasom) [reply src/status_im/ui/components/connectivity/view.cljs]: fixed

File: src/status_im/ui/components/connectivity/styles.cljs [hunk 2]
```diff
@@ -3,21 +3,23 @@
   (:require [status-im.ui.components.colors :as colors]
             [status-im.utils.platform :as platform]))
 
-(defnstyle text-wrapper [top opacity window-width pending?]
+(defnstyle text-wrapper [{:keys [top window-width pending? modal?]}]
   (cond->
-   {:opacity          opacity
+   {:opacity          1.0
```
Comment (flexsurfer) [inline src/status_im/ui/components/connectivity/styles.cljs:8]: just wondering why 1`.0` ?

File: src/status_im/ui/components/connectivity/view.cljs [hunk 3]
```diff
@@ -6,6 +6,22 @@
             [status-im.ui.components.connectivity.styles :as styles]
             [status-im.i18n :as i18n]))
 
+(defview error-label
+  [{:keys [view-id label mailserver-error?] :as opts}]
+
+  {:should-component-update
```
Comment (flexsurfer) [inline src/status_im/ui/components/connectivity/view.cljs:12]: @rasom  could you please elaborate ?  this fix looks complex , thanks

**Review by flexsurfer — APPROVED** [review]

**Review by goranjovic — APPROVED** [review]

File: src/status_im/ui/components/connectivity/styles.cljs (hunk 2 above)
Comment (rasom) [reply src/status_im/ui/components/connectivity/styles.cljs:8]: https://github.com/status-im/status-react/pull/6068/files#diff-df3f46463da5ee34e009067a466d7039L24 it was passed as a parameter before, idk

File: src/status_im/ui/components/connectivity/view.cljs (hunk 3 above)
Comment (rasom) [reply src/status_im/ui/components/connectivity/view.cljs:12]: https://github.com/status-im/status-react/pull/6068/files#diff-df3f46463da5ee34e009067a466d7039R17 at this line depending on `view-id` we pass `modal?` parameter to `styles/text-wrapper`. This `modal?` parameter affects `top` property of the style. 

When we navigate back from `:chat-modal` to any other screen `:view-id` in `app-db` gets changed while the animation is in progress, which causes the change of `top` position before the screen is hidden from the user. In result "Fetching messages..." is flickering. 

You can easily reproduce this on android device/emulator by removing `:should-component-update` handler.

**Comment by lukaszfryc** [comment]
@rasom works good to me. I tested on Galaxy S8 Android 8 and iPhone 8 iOS 12.

**Pull request merged**